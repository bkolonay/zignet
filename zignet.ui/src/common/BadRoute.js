import React, { Component } from 'react';

class BadRoute extends Component {

  render() {
    return (
      <p>Route does not exist</p>
    );
  }
}

export default BadRoute;
