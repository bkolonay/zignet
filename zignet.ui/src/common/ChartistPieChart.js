import React, { Component } from 'react';
import Chartist from 'chartist';
import './chartist.css';

class ChartistPieChart extends Component {

  componentDidMount() {
  	this.chartistChart = Chartist.Pie(
  		this.chartContainer,
  		this.props.chartData,
  		{ 
  			donut: true,
			donutWidth: 40
		}
	)
  }

  render() {
    return (
      <div 
        className="ct-chart" 
        id={"chart" + this.props.chartId}
        ref={(div) => this.chartContainer = div}
      />
    );
  }
}

export default ChartistPieChart;
