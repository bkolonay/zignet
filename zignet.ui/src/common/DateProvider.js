import moment from 'moment';

function now() {
	return moment();
}

export { now }