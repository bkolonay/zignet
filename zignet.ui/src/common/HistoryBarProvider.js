import moment from 'moment';

class HistoryBarProvider {

  /* 
    helped with percentage calculations:
     - https://www.calculatorsoup.com/calculators/math/percentage.php
     - https://www.timeanddate.com/date/timeduration.html
  */
  
	getErrorDivAttributes(historyBarWidth, now, testFailureDuration) {
		let oneDayAgo = moment(now.format()).subtract(24, 'hours');
    let testFailureStartTime = moment.utc(testFailureDuration.FailureStart).local();
    let minutesBetweenOneDayAgoAndTestFailureStartTime = moment.duration(testFailureStartTime.diff(oneDayAgo)).asMinutes();

    var failureDivStart = 0;
    if (minutesBetweenOneDayAgoAndTestFailureStartTime > 0) {
      let passingPercentOfLastDay = (minutesBetweenOneDayAgoAndTestFailureStartTime / 1440) * 100;
      failureDivStart = Math.floor((passingPercentOfLastDay / 100) * historyBarWidth);
    }
    else
      testFailureStartTime = oneDayAgo;

    var failureDivWidth = 0;
    if (testFailureDuration.FailureEnd) {
      let testFailureEndTime = moment.utc(testFailureDuration.FailureEnd).local();
      let failureDurationInMinutes = moment.duration(testFailureEndTime.diff(testFailureStartTime)).asMinutes();
      let failingPercentOfLastDay = (failureDurationInMinutes / 1440) * 100;
      failureDivWidth = Math.floor((failingPercentOfLastDay / 100) * historyBarWidth);
      if (failureDivWidth < 1)
      	failureDivWidth = 1;
    }
    else 
      failureDivWidth = historyBarWidth - failureDivStart;

    return {left: failureDivStart, width: failureDivWidth};
	}

  getDivTitle(testFailureDuration) {
    let testFailureStartTime = moment.utc(testFailureDuration.FailureStart).local();
    if (testFailureDuration.FailureEnd)
    {
      let testFailureEndTime = moment.utc(testFailureDuration.FailureEnd).local();
      let timeFromStartToEnd = testFailureStartTime.to(testFailureEndTime, true);
      return timeFromStartToEnd + ' from ' + testFailureStartTime.format('l h:mma') + ' - ' + testFailureEndTime.format('l h:mma');
    }
    else
    {
      let timeFromStartToEnd = testFailureStartTime.toNow(true);
      return timeFromStartToEnd + ' from ' + testFailureStartTime.format('l h:mma') + ' - now';
    }      
  }
}

export default HistoryBarProvider;