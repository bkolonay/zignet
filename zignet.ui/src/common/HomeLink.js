import React, { Component } from 'react';
import { getUrlParams } from '../common/routing/SearchFilter.js'

class HomeLink extends Component {

  render() {
    return (
      <a href={"/?" + getUrlParams(this.props.filter)}>ZigNet</a>
    );
  }
}

export default HomeLink;
