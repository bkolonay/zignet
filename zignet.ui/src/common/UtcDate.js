import moment from 'moment';

class UtcDate {
  constructor(date) {
    this.moment = moment.utc(date);
  }

  toLocal() {
    this.moment = this.moment.local();
    return this;
  }

  getDateAndTimeString() {
    return this.moment.format('L LTS');
  }

  getTimeFromNow() {
    return this.moment.fromNow(true);
  }

  getTimeFromNowWithSuffix() {
    return this.moment.fromNow();
  }  

  getTimeFrom(date) {
    const localMoment = moment.utc(date);
    return this.moment.from(localMoment, true);
  }  
}

export default UtcDate;
