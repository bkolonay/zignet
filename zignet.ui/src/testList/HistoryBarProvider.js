import React from 'react';
import moment from 'moment';

function getFailureDivs(testFailureDurations, historyBarDiv) {
  const historyBarWidth = historyBarDiv.offsetWidth;
  let now = moment();
  var failureDivs = [];
  for (var i = 0; i < testFailureDurations.length; i++) {
    let failureDivAttributes = getErrorDivAttributes(historyBarWidth, now, testFailureDurations[i]);
    failureDivs.push(<div key={i} style={failureDivAttributes} title={getDivTitle(testFailureDurations[i])}/>);
  }
  return failureDivs;
}

  /* 
    helped with percentage calculations:
     - https://www.calculatorsoup.com/calculators/math/percentage.php
     - https://www.timeanddate.com/date/timeduration.html
  */
  
function getErrorDivAttributes(historyBarWidth, now, testFailureDuration) {
	let oneDayAgo = moment(now.format()).subtract(24, 'hours');
  let testFailureStartTime = moment.utc(testFailureDuration.FailureStart).local();
  let minutesBetweenOneDayAgoAndTestFailureStartTime = moment.duration(testFailureStartTime.diff(oneDayAgo)).asMinutes();

  var failureDivStart = 0;
  if (minutesBetweenOneDayAgoAndTestFailureStartTime > 0) {
    let passingPercentOfLastDay = (minutesBetweenOneDayAgoAndTestFailureStartTime / 1440) * 100;
    failureDivStart = Math.floor((passingPercentOfLastDay / 100) * historyBarWidth);
  }
  else
    testFailureStartTime = oneDayAgo;

  var failureDivWidth = 0;
  if (testFailureDuration.FailureEnd) {
    let testFailureEndTime = moment.utc(testFailureDuration.FailureEnd).local();
    let failureDurationInMinutes = moment.duration(testFailureEndTime.diff(testFailureStartTime)).asMinutes();
    let failingPercentOfLastDay = (failureDurationInMinutes / 1440) * 100;
    failureDivWidth = Math.floor((failingPercentOfLastDay / 100) * historyBarWidth);
    if (failureDivWidth < 1)
    	failureDivWidth = 1;
  }
  else 
    failureDivWidth = historyBarWidth - failureDivStart;

  return {left: failureDivStart, width: failureDivWidth};
}

function getDivTitle(testFailureDuration) {
  let testFailureStartTime = moment.utc(testFailureDuration.FailureStart).local();
  if (testFailureDuration.FailureEnd)
  {
    let testFailureEndTime = moment.utc(testFailureDuration.FailureEnd).local();
    let timeFromStartToEnd = testFailureStartTime.to(testFailureEndTime, true);
    return timeFromStartToEnd + ' from ' + testFailureStartTime.format('l h:mma') + ' - ' + testFailureEndTime.format('l h:mma');
  }
  else
  {
    let timeFromStartToEnd = testFailureStartTime.toNow(true);
    return timeFromStartToEnd + ' from ' + testFailureStartTime.format('l h:mma') + ' - now';
  }      
}

export { getFailureDivs, getErrorDivAttributes, getDivTitle };