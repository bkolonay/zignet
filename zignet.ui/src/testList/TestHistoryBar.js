import React, { Component } from 'react';
import { getFailureDivs } from './HistoryBarProvider'
import { now } from '../common/DateProvider'
import './css/testHistory.css';

class TestHistoryBar extends Component {
  constructor() {
    super();
    this.state = {
      failureDivs: []
    };
  }

  componentDidMount() {
    this.setState({
      failureDivs: getFailureDivs(this.props.testFailureDurations, this.historyBarDiv, now())
    });
  }

  componentWillReceiveProps(props) {
    this.setState({
      failureDivs: getFailureDivs(this.props.testFailureDurations, this.historyBarDiv, now())
    });
  }

  render() {
    return (
      <div className="testHistoryBar" ref={(div) => this.historyBarDiv = div}>
        {this.state.failureDivs.map((failureDiv) => failureDiv)}      
      </div>
    );
  }
}

export default TestHistoryBar;