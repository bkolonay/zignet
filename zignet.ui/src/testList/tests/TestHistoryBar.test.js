import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestHistoryBar from '../TestHistoryBar';
import { getFailureDivs } from '../../common/HistoryBarProvider'

jest.mock('../../common/HistoryBarProvider');

it('renders with no failures', () => {
  getFailureDivs.mockReturnValue([]);
  const testFailureDurations = [];

  const component = <TestHistoryBar testFailureDurations={testFailureDurations}/>;
  ReactDOM.render(component,document.createElement('div'));
});

it('renders with single failure duration', () => {
  getFailureDivs.mockReturnValue([<div key={0}/>]);
  const testFailureDurations = [{
    FailureStart: '2018-05-10T01:00:00-07:00',
    FailureEnd: '2018-05-10T02:00:00-07:00'
  }];

  const component = <TestHistoryBar testFailureDurations={testFailureDurations}/>;
  ReactDOM.render(component,document.createElement('div'));
});

it('renders with multiple failure durations', () => {
  getFailureDivs.mockReturnValue([<div key={0}/>, <div key={1}/>]);
  const testFailureDurations = [{
    FailureStart: '2018-05-10T01:00:00-07:00',
    FailureEnd: '2018-05-10T02:00:00-07:00'
  },
  {
    FailureStart: '2018-05-10T05:00:00-07:00',
    FailureEnd: '2018-05-10T07:00:00-07:00'
  }];

  const component = <TestHistoryBar testFailureDurations={testFailureDurations}/>;
  ReactDOM.render(component,document.createElement('div'));
});

it('snapshot with no failures', () => {
  getFailureDivs.mockReturnValue([]);
  const testFailureDurations = [];

  const component = <TestHistoryBar testFailureDurations={testFailureDurations}/>;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();  
});

it('snapshot with single failure duration', () => {
  getFailureDivs.mockReturnValue([<div key={0} id="div0"/>]);
  const testFailureDurations = [{
    FailureStart: '2018-05-16T01:00:00-07:00',
    FailureEnd: '2018-05-16T02:00:00-07:00'
  }];

  const component = <TestHistoryBar testFailureDurations={testFailureDurations}/>;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot with a multiple failure durations', () => {
  getFailureDivs.mockReturnValue([<div key={0} id="div0"/>, <div key={1} id="div1"/>]);
  const testFailureDurations = [{
    FailureStart: '2018-05-16T01:00:00-07:00',
    FailureEnd: '2018-05-16T02:00:00-07:00'
  },
  {
    FailureStart: '2018-05-16T05:00:00-07:00',
    FailureEnd: '2018-05-16T07:00:00-07:00'
  }];

  const component = <TestHistoryBar testFailureDurations={testFailureDurations}/>;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});