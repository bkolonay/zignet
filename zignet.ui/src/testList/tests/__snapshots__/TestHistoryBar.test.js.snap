// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot with a multiple failure durations 1`] = `
<div
  className="testHistoryBar"
>
  <div
    id="div0"
  />
  <div
    id="div1"
  />
</div>
`;

exports[`snapshot with no failures 1`] = `
<div
  className="testHistoryBar"
/>
`;

exports[`snapshot with single failure duration 1`] = `
<div
  className="testHistoryBar"
>
  <div
    id="div0"
  />
</div>
`;
